### PR TITLE
convert old release notes to md

### PR DESCRIPTION
**Pull request by sofroniewn**
# Description
Following #1043 this PR converts all our old release notes from `rst` to `md`, adjusts headers to be the same, and puts contributors names in with the new formatting. It changes none of the content of the old release notes only fixes these formatting issues so that from now on release notes will be consistent.

I still have to update a few contributor names in some of our old releases, will do that before merge (hit a rate limit on the github api right now), but this is otherwise ready for review.

We might need to adjust our docs so that these render correctly. Maybe @kne42 or @tlambert03 can help with that.

### CODE REVIEW

**Comment by sofroniewn** [comment]
Github handles have now been updated, this is ready for review

File: docs/release/generate_release_notes.py [hunk 1]
```diff
@@ -125,7 +125,7 @@ def add_to_users(users, new_user):
 highlights['Highlight'] = {}
 highlights['New Feature'] = {}
 highlights['Improvement'] = {}
-highlights['Bugfix'] = {}
+highlights['Bug Fixe'] = {}
```
Comment (kne42) [inline docs/release/generate_release_notes.py]: typo

**Review by kne42 — APPROVED** [review]

File: docs/release/generate_release_notes.py (hunk 1 above)
Comment (sofroniewn) [reply docs/release/generate_release_notes.py]: good catch, thanks!

**Comment by sofroniewn** [comment]
Typo fixed! Will merge after tests pass

**Pull request merged**